### PR TITLE
Recyclerview v3

### PR DESCRIPTION
**Pull request by nbhd**
https://github.com/nbhd/welcome-to-android/issues/11
- [x] @shaunkawano 
- [x] @eoinfogarty 
- [x] @bowyer-app

### CODE REVIEW

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java [hunk 1]
```diff
@@ -74,11 +103,14 @@ public int getItemCount() {
     }
 
     class ItemViewHolder extends RecyclerView.ViewHolder {
+
         public TextView textView;
         public ImageView imageView;
+        public View itemView;
```
Comment (eoinfogarty) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java]: you dont need to do this. 
this is done automatically.

Comment (shaunkawano) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java]: Did you need to declare this? I think it's already accessible through `ViewHolder`.

Comment (nbhd) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java]: super class already has itemView 🙇 
I don't read carefully super class.
thanks!

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package net.boondockradio.recyclerview;
+
+public interface StatesDelegate {
+
+    public void remove(int position);
```
Comment (bowyer-app) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java]: this class is unnecessary.

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package net.boondockradio.recyclerview;
+
+public interface StatesDelegate {
```
Comment (eoinfogarty) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java]: normally we call this a `Listener` or `Callback`

`StateAdapterListener` or `StateAdapterCallback`

Comment (bowyer-app) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java]: I wanted to say the same thing

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java [hunk 4]
```diff
@@ -66,6 +70,31 @@ public void onClick(View v) {
...
+
+        holder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
+            @Override
+            public boolean onLongClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(context);
+                builder
+                        .setMessage(
+                                context.getString(
+                                        R.string.dialog_remove_message,
+                                        states.get(holder.getAdapterPosition()).title
```
Comment (eoinfogarty) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java]: can use `item.title`?

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java (hunk 3 above)
Comment (nbhd) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java]: 👍

Comment (bowyer-app) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesDelegate.java]: In general, `Listener` or `Callback` use as an inner class.
http://qiita.com/ushi3_jp/items/ba0df2df6b0b77278f14

File: RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java (hunk 4 above)
Comment (nbhd) [inline RecyclerView/app/src/main/java/net/boondockradio/recyclerview/StatesAdapter.java]: oops

File: RecyclerView/app/src/main/res/layout/dialog_add_state.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <EditText
+        android:id="@+id/stateName"
```
Comment (shoheikawano) [inline RecyclerView/app/src/main/res/layout/dialog_add_state.xml]: this is ok, but it is common to apply snake case naming convention for xml ids maybe

**Comment by shoheikawano** [comment]
nice! LGTM!

LGTM

**Comment by eoinfogarty** [comment]
LGTM

**Comment by bowyer-app** [comment]
LGTM

**Pull request merged**